x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/token/InvalidateTokenResponse.java

```diff
@@ -5,41 +5,75 @@
  */
 package org.elasticsearch.xpack.core.security.action.token;
 
+import org.elasticsearch.Version;
 import org.elasticsearch.action.ActionResponse;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.core.security.authc.support.TokensInvalidationResult;
 
 import java.io.IOException;
+import java.util.Objects;
 
 /**
- * Response for a invalidation of a token.
+ * Response for a invalidation of one or multiple tokens.
  */
-public final class InvalidateTokenResponse extends ActionResponse {
+public final class InvalidateTokenResponse extends ActionResponse implements ToXContent {
 
-    private boolean created;
+    private TokensInvalidationResult result;
 
     public InvalidateTokenResponse() {}
 
-    public InvalidateTokenResponse(boolean created) {
-        this.created = created;
+    public InvalidateTokenResponse(TokensInvalidationResult result) {
+        this.result = result;
     }
 
-    /**
-     * If the token is already invalidated then created will be <code>false</code>
-     */
-    public boolean isCreated() {
-        return created;
+    public TokensInvalidationResult getResult() {
+        return result;
+    }
+
+    private Boolean isCreated() {
+        return result.getInvalidatedTokens().size() > 0
+            && result.getPreviouslyInvalidatedTokens().isEmpty()
+            && result.getErrors().isEmpty();
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
-        out.writeBoolean(created);
+        if (out.getVersion().before(Version.V_7_0_0)) {
+            out.writeBoolean(isCreated());
+        } else {
+            TokensInvalidationResult.writeTo(result, out);
+        }
     }
 
     @Override
     public void readFrom(StreamInput in) throws IOException {
         super.readFrom(in);
-        created = in.readBoolean();
+        if (in.getVersion().before(Version.V_7_0_0)) {
+            result = TokensInvalidationResult.emptyResult();
```

Comment:
hmm shouldn't you read the boolean here? The older versions are going to send it so we should read it and ensure isCreated() would return the right value

Reply:
> hmm shouldn't you read the boolean here? The older versions are going to send it so we should read it

I was thinking of this as a NOOP of sorts (hence the `emtpyResult()` ). I could also build a 

- TokensAuthetnicationResult with one random string in`invalidatedTokens` and an empty list for `previouslyInvalidatedTokens` when created is `true`
- TokensAuthetnicationResult with one random string in`previouslyInvalidatedTokens` and an empty list for `invalidatedTokens` when created is `false`

in order to simulate the result better. 

> and ensure isCreated() would return the right value
We don't do that now either, right? Maybe I'm missing what you're alluding to, though :)